File: sample-test-001/code.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;



class Result {

    /*
     * Complete the 'fizzBuzz' function below.
     *
     * The function accepts INTEGER n as parameter.
     */

    public static void fizzBuzz(int n) {
    // Write your code here
        final String FIZZ = "Fizz";
        final String BUZZ = "Buzz";
        final Function<Integer, Boolean> divisibleByThree = (i) -> 0 == i % 3;
        final Function<Integer, Boolean> divisibleByFive = (i) -> 0 == i % 5;
        if (divisibleByThree.apply(n) && divisibleByThree.apply(n)) {
            System.out.println(FIZZ + BUZZ);
        } else if (divisibleByFive.apply(n)) {
            System.out.println(FIZZ);
        } else if (divisibleByFive.apply(n)) {
            System.out.println(BUZZ);
        } else {
            System.out.println(n);
        }
    }

}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        Result.fizzBuzz(n);

        bufferedReader.close();
    }
}

File: src/main/java/training01/data_structures/arrays/a02_2d_array_ds/code.scala
import java.io._
import java.math._
import java.security._
import java.text._
import java.util._
import java.util.concurrent._
import java.util.function._
import java.util.regex._
import java.util.stream._

object Solution0002 {

    // Complete the hourglassSum function below.
    def hourglassSum(arr: Array[Array[Int]]): Int = {
        if (null == arr || 0 == arr.length || 0 == arr(0).length)
            0
        (1 to 4).toList
            .flatMap(hourglassNumber => (1 to 4).toList
                .map(index => arr.calculateSum(hourglassNumber, index)))
                .max
    }

    implicit class ArrayExtensions(arr: Array[Array[Int]]) {
        def calculateSum(horizontal: Int, vertical: Int): Int = {
            arr(vertical)(horizontal) +
            arr(vertical-1)(horizontal-1) +
            arr(vertical-1)(horizontal) +
            arr(vertical-1)(horizontal+1) +
            arr(vertical+1)(horizontal-1) +
            arr(vertical+1)(horizontal) +
            arr(vertical+1)(horizontal+1)
        }
    }

    def main(args: Array[String]) {
        val stdin = scala.io.StdIn

        val printWriter = new PrintWriter(sys.env("OUTPUT_PATH"))

        val arr = Array.ofDim[Int](6, 6)

        for (i <- 0 until 6) {
            arr(i) = stdin.readLine.split(" ").map(_.trim.toInt)
        }

        val result = hourglassSum(arr)

        printWriter.println(result)

        printWriter.close()
    }
}
